### PR TITLE
feat(chip): add element and clear button

### PR DESCRIPTION
**Pull request by andresin87**
# add element and clear button

**TASK**: 

### Types of changes

- [x] ✨ New feature (non-breaking change which adds functionality)

### CODE REVIEW

File: packages/components/chip/src/Chip.styles.tsx [hunk 1]
```diff
@@ -0,0 +1,114 @@
+import { makeVariants, tw } from '@spark-ui/internal-utils'
+import { cva, type VariantProps } from 'class-variance-authority'
+
+// import { dashedVariants, filledVariants, outlinedVariants, tintedVariants } from './variants'
```
Comment (andresin87) [inline packages/components/chip/src/Chip.styles.tsx:4]: commented to be added on a new PR with stories

File: packages/components/chip/src/Chip.styles.tsx [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+      }),
+      /**
+       * Disable the chip, preventing user interaction and adding opacity.
+       */
+      disabled: {
+        true: ['cursor-not-allowed', 'opacity-dim-3'],
+      },
+    },
+    compoundVariants: [
+      // ...filledVariants,
```
Comment (andresin87) [inline packages/components/chip/src/Chip.styles.tsx:56]: commented to be on a new PR

File: packages/components/chip/src/Chip.styles.tsx [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+    compoundVariants: [
+      {
+        hasClearButton: false,
+        isBordered: 'no',
+        class: tw(['px-md']),
+      },
+      {
+        hasClearButton: false,
+        isBordered: 'yes',
+        class: tw(['px-[7px]']),
```
Comment (andresin87) [inline packages/components/chip/src/Chip.styles.tsx:94]: it will be improved on next PR

Comment (andresin87) [reply packages/components/chip/src/Chip.styles.tsx:94]: just for stiling

File: packages/components/chip/src/Chip.tsx [hunk 4]
```diff
@@ -1,7 +1,105 @@
...
+  return element ? (element.type as FC).displayName : ''
+}
+
+const findElement =
+  (...values: string[]) =>
+  (children: React.ReactNode) => {
+    const validChildren = Children.toArray(children).filter(isValidElement)
+
+    return validChildren.find(child => values.includes(getDisplayName(child) || ''))
+  }
```
Comment (andresin87) [inline packages/components/chip/src/Chip.tsx:25]: Taken from Input
I think that part might be moved as a util in internal-utils
What do you think @andresz1 ?

**Comment by Powerplex** [comment]
@andresin87 do you know why there is no test env deployed for your PRs about the Chip ?

File: packages/components/chip/src/Chip.tsx [hunk 5]
```diff
@@ -1,7 +1,105 @@
...
+          intent,
+        })}
+        {...{
+          ...chipProps,
+          ...otherProps,
+        }}
+        data-spark-component="chip"
+      >
+        <span className={chipContentStyles({ hasClearButton: !!hasClearButton })}>
+          <ChipContext.Provider value={{ disabled, design, intent }}>
```
Comment (Powerplex) [inline packages/components/chip/src/Chip.tsx:96]: Maybe it is just a habit on my side but I tend to put the context providers high in the markup (around `ChipElement`)

File: packages/components/chip/src/ChipClearButton.tsx [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+      <span
+        className={chipClearButtonWrapperStyles({
+          isBordered: design === 'dashed',
+          disabled: !!disabled,
+        })}
+        onClick={onClearHandler}
+      >
+        <button
+          type="button"
+          {...{ ...(disabled && { disabled }) }}
```
Comment (Powerplex) [inline packages/components/chip/src/ChipClearButton.tsx:44]:
```
{ ...(disabled && { disabled }) }
```

Could be:
```
{...{ ...(disabled && { disabled }) }}
```

File: packages/components/chip/src/useChipElement.tsx [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+    | {
+        type: 'button'
+        'aria-pressed'?: boolean
+        'data-state'?: 'on' | 'off'
+        onClick: (event: React.MouseEvent<HTMLButtonElement>) => void
+        disabled?: true
+      }
+    | {
+        'aria-disabled'?: true
+      }
```
Comment (acd02) [inline packages/components/chip/src/useChipElement.tsx:38]: To improve legibility, maybe we could extract that type into its own variable., e.g.:
```tsx
interface ReturnedValue {
  Element: ...,
  chipProps: ...
}
```

File: packages/components/chip/src/ChipClearButton.tsx (hunk 6 above)
[the remark this answers is not in the document]
Comment (andresin87) [reply packages/components/chip/src/ChipClearButton.tsx]: this was for nod adding extra markup not needed

File: packages/components/chip/src/useChipElement.tsx (hunk 7 above)
Comment (Powerplex) [reply packages/components/chip/src/useChipElement.tsx:38]: or 

```
type ChipComputedProps = {
        type: 'button'
        'aria-pressed'?: boolean
        'data-state'?: 'on' | 'off'
        onClick: (event: React.MouseEvent<HTMLButtonElement>) => void
        disabled?: true
      }
    | {
        'aria-disabled'?: true
      }
```

then use this type in the interface (I just like the idea of making it explicit that the hook is computing the prop for the chip ^^") but its pretty similar
```
interface ChipContextValue {
  Element: 
}
```

**Review by acd02 — APPROVED** [review]

File: packages/components/chip/src/Chip.tsx (hunk 4 above)
Comment (andresin87) [reply packages/components/chip/src/Chip.tsx:25]: where and how it must be named
[directory-name]/[filename].ts

**Pull request merged**